### PR TITLE
add helper & fix logic for users show

### PR DESCRIPTION
**Pull request by chrislabarge**
User UI improvements (fixes?) #950

Added some user feature specs to show desired logic.

### CODE REVIEW

**Comment by chrislabarge** [comment]
I don't believe any of the errors on Travis are a result from my code, as they are all showing up on my local master branch as well. Please correct me if I am wrong. :)

**Comment by Br3nda** [comment]
@chrislabarge can you merge in the latest gem file fixes.?

File: app/helpers/users_helper.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module UsersHelper
+  def list_gardens_on_show?
+    has_non_default_garden?(@user) || authorized_for?(@user)
+  end
+
+  def has_non_default_garden?(user)
+    user.gardens.each { |garden| return true unless garden.user_default?}
```
Comment (Heirian) [inline app/helpers/users_helper.rb]: that line can be this way? `user.gardens.none?(&:user_default?)`

**Comment by RickCarlino** [comment]
Closing this PR due to inactivity, although it appears that RSpec is working again after upgrades. Sorry for the radio silence on this one- I'm hoping to take a more active role in getting things merged in 2019. Please let me know if there is anything I can do to help- I realize it has been very quiet these days.

**Pull request closed without merging**